### PR TITLE
feat(cognito): user pool identity support for Google

### PR DESCRIPTION
**Pull request by asjadsaboor-10p**
Added support for Google  in cognito user pool identity.

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Review by ubaidbaig-10p — APPROVED** [review]

File: packages/@aws-cdk/aws-cognito/lib/user-pool-idps/base.ts [hunk 1]
```diff
@@ -36,6 +36,25 @@ export class ProviderAttribute {
...
   public static readonly FACEBOOK_LOCALE = new ProviderAttribute('locale');
 
+  /** The name attribute provided by Google */
+  public static readonly GOOGLE_NAMES = new ProviderAttribute('names');
+  /** The gender attribute provided by Google */
+  public static readonly GOOGLE_GENDER = new ProviderAttribute('gender');
+  /** The birthday attribute provided by Google */
+  public static readonly GOOGLE_BIRTHDAY = new ProviderAttribute('birthdays');
+  /** The birthday attribute provided by Google */
+  public static readonly GOOGLE_PHONE_NUMBER = new ProviderAttribute('phoneNumbers');
```
Comment (nija-at) [inline packages/@aws-cdk/aws-cognito/lib/user-pool-idps/base.ts:46]: Use plural here as well.

```suggestion
  /** The birthdays attribute provided by Google */
  public static readonly GOOGLE_BIRTHDAYS = new ProviderAttribute('birthdays');
  /** The phone numbers attribute provided by Google */
  public static readonly GOOGLE_PHONE_NUMBERS = new ProviderAttribute('phoneNumbers');
```

**Review by nija-at — DISMISSED** [review]
Thanks for submitting this PR. It looks great.

Could you write a CDK app similar to this but with the google IDP - https://github.com/aws/aws-cdk/blob/master/packages/%40aws-cdk/aws-cognito/test/integ.user-pool-idp.ts - and confirm that the steps detailed at the top of the file works?

**Comment by nija-at** [comment]
cc @shivlaks

**Comment by asjadsaboor** [comment]
@nija-at I am having little trouble in runing the integ tests, getting the following error
`21/34 | 10:33:31 AM | CREATE_FAILED        | AWS::Cognito::UserPool       | myuserpool (myuserpool01998219) Cannot configure From email address for default email configuration (Service: AWSCognitoIdentityProviderService; Status Code: 400; Error Code: InvalidParameterException; Request ID: 1b48a4be-4d58-4be2-9d9b-101f77a4b769; Proxy: null)
	new UserPool (/home/asjad/development/projects/aws-cdk/packages/@aws-cdk/aws-cognito/lib/user-pool.js:191:26)
	\_ Object.<anonymous> (/home/asjad/development/projects/aws-cdk/packages/@aws-cdk/aws-cognito/test/integ.user-pool-explicit-props.js:8:18)
	\_ Module._compile (internal/modules/cjs/loader.js:1201:30)
`
 seems like the root cause is the email addresses used . If you can help me with this ?

`"EmailConfiguration": {
          "From": "noreply@myawesomeapp.com",
          "ReplyToEmailAddress": "support@myawesomeapp.com"
        },`

**Comment by nija-at** [comment]
@asjadsaboor - does it work if you remove the `From` property in the `integ.user-pool-explicit-props.ts` integration test?

**Comment by asjadsaboor** [comment]
@nija-at , yes it worked after removing it.  I have pushed all feedback items.

File: packages/@aws-cdk/aws-cognito/test/integ.user-pool-domain-cfdist.expected.json [hunk 2]
```diff
@@ -203,17 +207,17 @@
...
+    "AssetParametersb64b129569a5ac7a9abf88a18ac0b504d1fb1208872460476ed3fd435830eb94S3VersionKeyCCDC67C0": {
       "Type": "String",
-      "Description": "S3 key for asset version \"d731b1475f16a318a48a76c83d255f7422cfa5f025c5bff93537b8f0b8e94557\""
+      "Description": "S3 key for asset version \"b64b129569a5ac7a9abf88a18ac0b504d1fb1208872460476ed3fd435830eb94\""
     },
-    "AssetParametersd731b1475f16a318a48a76c83d255f7422cfa5f025c5bff93537b8f0b8e94557ArtifactHash5701DE73": {
+    "AssetParametersb64b129569a5ac7a9abf88a18ac0b504d1fb1208872460476ed3fd435830eb94ArtifactHash782948FC": {
       "Type": "String",
-      "Description": "Artifact hash for asset \"d731b1475f16a318a48a76c83d255f7422cfa5f025c5bff93537b8f0b8e94557\""
+      "Description": "Artifact hash for asset \"b64b129569a5ac7a9abf88a18ac0b504d1fb1208872460476ed3fd435830eb94\""
```
Comment (nija-at) [inline packages/@aws-cdk/aws-cognito/test/integ.user-pool-domain-cfdist.expected.json:220]: Why are these changing? I don't see any changes to user pool domain. These asset hashes shouldn't change.

Can you take a look?

**Review by nija-at — CHANGES_REQUESTED** [review]

File: packages/@aws-cdk/aws-cognito/test/integ.user-pool-domain-cfdist.expected.json (hunk 2 above)
Comment (asjadsaboor) [reply packages/@aws-cdk/aws-cognito/test/integ.user-pool-domain-cfdist.expected.json:220]: I executed the integration tests which added/updates these changes. Should I remove changes in this file ?

**Review by nija-at — CHANGES_REQUESTED** [review]
mergify dismissed my review. Moving this back to 'request changes'.
See comments above.

File: packages/@aws-cdk/aws-cognito/test/integ.user-pool-idp.amazon.ts [hunk 3]
```diff
@@ -7,7 +7,7 @@ import { ProviderAttribute, UserPool, UserPoolIdentityProviderAmazon } from '../
  * * If you plug in valid 'Login with Amazon' credentials, the federated log in should work.
  */
 const app = new App();
```
Comment (nija-at) [inline packages/@aws-cdk/aws-cognito/test/integ.user-pool-idp.amazon.ts:9]: missed this in this last iteration. I believe the corresponding expectation file - `integ.user-pool-idp.expected.json` - needs to be deleted.

File: packages/@aws-cdk/aws-cognito/test/integ.user-pool-explicit-props.ts [hunk 4]
```diff
@@ -76,7 +72,7 @@ const userpool = new UserPool(stack, 'myuserpool', {
 
 const cognitoDomain = userpool.addDomain('myuserpooldomain', {
   cognitoDomain: {
-    domainPrefix: 'myawesomeapp',
+    domainPrefix: 'myawesomeapp2',
```
Comment (nija-at) [inline packages/@aws-cdk/aws-cognito/test/integ.user-pool-explicit-props.ts:75]: can we revert this back?

File: packages/@aws-cdk/aws-cognito/test/integ.user-pool-explicit-props.expected.json [hunk 5]
```diff
@@ -850,7 +848,7 @@
     "myuserpoolmyuserpooldomainEE1E11AF": {
       "Type": "AWS::Cognito::UserPoolDomain",
       "Properties": {
-        "Domain": "myawesomeapp",
+        "Domain": "myawesomeapp2",
```
Comment (nija-at) [inline packages/@aws-cdk/aws-cognito/test/integ.user-pool-explicit-props.expected.json:851]: any reason you've made this change?

**Review by nija-at — CHANGES_REQUESTED** [review]
Last few small comments I missed in the previous round. Shouldbe good to go after this.

File: packages/@aws-cdk/aws-cognito/test/integ.user-pool-explicit-props.expected.json (hunk 5 above)
Comment (asjadsaboor) [reply packages/@aws-cdk/aws-cognito/test/integ.user-pool-explicit-props.expected.json]: It was conflicting with another test , I have reverted it back.

**Review by nija-at — APPROVED** [review]

**Pull request merged**